### DISCUSSION TITLE
Vue3 微信小程序 definePageConfig 不能读取配置文件

### DISCUSSION THREAD

**Discussion by GiantappMan** [NervJS/taro#11704] [Feature Request]
不行
```
const page = "首页";
definePageConfig({
  navigationBarTitleText: page,
});
```
---
行
```
definePageConfig({
  navigationBarTitleText: "首页",
});
```

我尝试动态读取标题，但是发现不行。写死字符串就可以，请问这是为什么。。。

**Comment by GiantappMan** [comment]
我希望把所有文本定义到配置中，方便直接修改

**Comment by b2nil** [comment]
@DaZiYuan 
SFC 中的 `definePageConfig` 目前是用正则匹配读取配置的，无法处理参数为变量或含有变量的情况。
如果必须要使用变量的话，暂时可以用单独的配置文件来处理。

**Reply by GiantappMan** [reply]
> @DaZiYuan SFC 中的 `definePageConfig` 目前是用正则匹配读取配置的，无法处理参数为变量或含有变量的情况。 如果必须要使用变量的话，暂时可以用单独的配置文件来处理。

单独配置文件是哪种方式，麻烦详细解释一下。。。感谢
我这里单独弄了个ts来存，读取也不行 
```
import {data} from 'xxx'
console.log(data);//有值，
definePageConfig({
  navigationBarTitleText:data, //还是没值
});

```

**Reply by b2nil** [reply]
单独的页面配置文件，详见[文档](https://taro-docs.jd.com/taro/docs/page-config#%E5%9C%A8%E9%A1%B5%E9%9D%A2%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6%E4%B8%AD%E4%BD%BF%E7%94%A8)。

页面配置需要使用 `export default` 把配置暴露给 `tarojs/helper` 中的 `readConfig` 来读取：
```ts
import {data} from 'xxx'
console.log(data);//有值，
export default definePageConfig({
  navigationBarTitleText:data, 
});
```

**Reply by GiantappMan** [reply]
可以了，但是别名似乎在单独配置中不生效，相对路径才可以

**Reply by b2nil** [reply]
是的，目前 taro 分析页面配置文件中的引入路径时，暂时是没有处理别名的。兄弟有空的话，可以考虑考虑提个 Issue 或 PR。
https://github.com/NervJS/taro/blob/71c900fc056bba711505ab5bb605dbe973df2413/packages/taro-helper/src/utils.ts#L494-L534

**Comment by GiantappMan** [comment]
https://github.com/NervJS/taro/issues/11716  提交了一个Issue
